utils/questGen.py

```diff
@@ -9,71 +17,165 @@ def generate_quest(quest):
-            if '{0}' in quest_type:
-                quest_type_text = quest_type.replace('{0}', str(quest['quest_target']))
-                quest_target = str(quest['quest_target'])
-        elif quest_reward_type == 'Pokemon':
+        elif quest_reward_type == _("Pokemon"):
             item_amount = "1"
             item_type = "Pokemon"
             item_id = "000"
-            pokemon_name = pokemonname(str(quest['quest_pokemon_id']))
+            pokemon_name = i8ln(pokemonname(str(quest['quest_pokemon_id'])))
             pokemon_id = str(quest['quest_pokemon_id'])
-            if '{0}' in quest_type:
-                quest_type_text = quest_type.replace('{0}', str(quest['quest_target']))
-                quest_target = str(quest['quest_target'])
 
+        if '{0}' in quest_type:
+            quest_type_text = quest_type.replace('{0}', quest_target)
+
+        quest_task = questtask(quest_type_raw, quest_condition, quest_target)
         
         quest_raw = ({'pokestop_id': pokestop_id, 'latitude': latitude, 'longitude': longitude, 
-            'quest_type_raw': quest_type, 'quest_type': quest_type_text, 'item_amount': item_amount, 'item_type': item_type, 
+            'quest_type_raw': quest_type_raw, 'quest_type': quest_type_text, 'quest_condition': quest_condition, 'item_amount': item_amount, 'item_type': item_type, 
             'quest_target': quest_target, 'name': name, 'url': url, 'timestamp': timestamp, 'pokemon_id': pokemon_id, 'item_id': item_id,
-            'pokemon_name': pokemon_name, 'quest_reward_type': quest_reward_type, 'quest_reward_type_raw': quest_reward_type_raw})
+            'pokemon_name': pokemon_name, 'quest_reward_type': quest_reward_type, 'quest_reward_type_raw': quest_reward_type_raw, 'quest_task': quest_task})
         return quest_raw
     
 def questreward(quest_reward_type):
     type = {
-        2: "Item",
-        3: "Stardust",
-        7: "Pokemon"
+        2: _("Item"),
+        3: _("Stardust"),
+        7: _("Pokemon")
     }
     return type.get(quest_reward_type, "nothing")
     
 def questtype(quest_type):
-    with open('utils/quest/types.json') as f:
-        items = json.load(f)
-    return (items[str(quest_type)]['text'])
+    file = open_json_file('types')
+    return (file[str(quest_type)]['text'])
     
 def rewarditem(itemid):
-    with open('utils/quest/items.json') as f:
-        items = json.load(f)
-    return (items[str(itemid)]['name'])
+    file = open_json_file('items')
+    return (file[str(itemid)]['name'])
     
 def pokemonname(id):
-    with open('pokemon.json') as f:
-        mondata = json.load(f)
-    return mondata[str(int(id))]["name"]
+    file = open_json_file('pokemon')
+    return file[str(int(id))]["name"]
+
+def questtask(typeid, condition, target):
+    pokemonTypes = open_json_file('pokemonTypes')
+    items = open_json_file('items')
+    throwTypes = {"10":_("Nice"), "11":_("Great"), "12":_("Excellent"), "13":_("Curveball")}
```

Comment:
@georgeherby 
For consistency of having all this stuff in JSON file, I would expect this to be in a JSON file. Just easier to update if Nia make a change on throws for example